ts/Series/FlowMap/FlowMapSeries.ts

```diff
@@ -0,0 +1,834 @@
+                    fromY + fromYToArc
+                ],
+                ['Z']
+            ]
+        };
+
+        if (markerEndOptions && markerEndOptions.enabled && shapeArgs.d) {
+            const marker: SVGPath = FlowMapSeries.markerEndPath(
+                [toX - toXToArc, toY - toYToArc],
+                [toX + toXToArc, toY + toYToArc],
+                [toPoint.plotX as number, toPoint.plotY as number],
+                markerEndOptions
+            );
+
+            (shapeArgs.d as SVGPath).splice(2, 0,
+                ...marker
+            );
+        }
+
+        return shapeArgs;
+    }
+}
+
+/* *
+ *
+ *  Prototype properties
+ *
+ * */
+
+interface FlowMapSeries {
+    pointClass: typeof FlowMapPoint;
+}
+
+extend(FlowMapSeries.prototype, {
+    pointClass: FlowMapPoint,
+    // Make it work on zoom or pan
+    useMapGeometry: true
+});
+
+/* *
+ *
+ *  Registry
+ *
+ * */
+declare module '../../Core/Series/SeriesType' {
+    interface SeriesTypeRegistry {
+        flowmap: typeof FlowMapSeries;
+    }
+}
+
+SeriesRegistry.registerSeriesType('flowmap', FlowMapSeries);
+
+/* *
+ *
+ *  Default export
+ *
+ * */
+
+export default FlowMapSeries;
+
+/* *
+ *
+ *  API options
+ *
+ * */
+
+/**
+ * A `flowmap` series. If the [type](#series.flowmap.type) option
+ * is not specified, it is inherited from [chart.type](#chart.type).
+ *
+ * @extends   series,plotOptions.flowmap
+ * @excluding allAreas, allowPointSelect, boostBlending, boostThreshold,
+ * borderColor, borderWidth, dashStyle, dataLabels, dragDrop, joinBy, mapData,
+ * negativeColor, onPoint, shadow, showCheckbox
+ * @product   highmaps
+ * @apioption series.flowmap
+ */
+
+/**
+ * An array of data points for the series. For the `flowmap` series
+ * type, points can be given in the following ways:
+ *
+ * 1.  An array of arrays with 6 values. In this case, the values correspond
+ *     to `from, to, weight`. Example:
+ *     ```js
+ *     data: [
+ *         ["Point 1", "Point 2", 4],
+ *     ]
+ *     ```
+ *
+ * 2.  An array of objects with named values. The following snippet shows only a
+ *     few settings, see the complete options set below.
+ *
+ *     ```js
+ *         data: [{
+ *             from: "Point 1",
+ *             to: "Point 2",
+ *             curveFactor: 0.4,
+ *             weight: 5,
+ *             growTowards: true,
+ *             markerEnd: {
+ *                 enabled: true,
+ *                 height: 15,
+ *                 width: 8
+ *             }
+ *         }]
+ *     ```
+ *
+ * @type      {Array<number|null|*>}
+ * @excluding labelrank, outgoing
+ * @apioption series.flowmap.data
+ */
+
+/**
+ * Higher numbers makes the links more curved. A `curveFactor` of 0 makes the
+ * lines straight.
+ *
+ * @sample {highmaps} maps/demo/flowmap-ship-route/
+ *         Example ship route
+ *
+ * @type      {number}
+ * @apioption series.flowmap.data.curveFactor
+ */
+
+/**
+ * The fill color of an individual link
```

Comment:
In general, there is some inconsistency when it comes to code comments, and some jsdocs too.
I didn't want to mark every one, but mainly it's about some missing dots at the end of sentences.